src/util/getBreakpoints.js

```diff
@@ -30,9 +38,19 @@ function getBreakpoints(store) {
     }
 
     // return the breakpoints in the redux store
-    return storeState['@@__IMMUTABLE_ITERABLE__@@']
-            ? storeState.get(responsiveStateKey).breakpoints
-            : storeState[responsiveStateKey].breakpoints
+
+    //	first check/handle if Immutable is being used
+    if (storeState['@@__IMMUTABLE_ITERABLE__@@']) {
+        const responsiveState = storeState.get(responsiveStateKey)
+        // check if Immtable Map is being used, requiring .get()
+        if (responsiveState['@@__IMMUTABLE_MAP__@@']) {
```

Comment:
I don't believe `responsiveState` will ever be an Immutable type - `redux-responsive`'s `createReducer` always returns plain object state.

Even if the user of this lib is using `immutable` and `redux-immutable`, `redux-immutable` only sets the root of the store as an Immutable type - all other branches of state have their types controlled by their respective reducers (and aren't necessarily immutable types). So in our case, the "responsive" branch of state will always be a plain object, because that's what its reducer returns.

Reply:
I think it was a bit more complicated than I originally thought & explained.  It looks like the scenario I am hitting is when we use preloadedState between server and client.  When the client creates the store from the JSON of preloaded state from the server, we first convert it the whole preloadedState into an Immutable object.

Here is an example of a test that demonstrates the scenario:
```javascript
it('Can find reducer at root after createStore with preloadedState (immutable)', function() {
    const reducers = immutableCombine({
        other: () => { return {testKey: 'reducedValue'}},
        browser: reducer
    })
    const preloadedState = Immutable.fromJS({
        other: {
            testKey: 'preloadedValue',
        },
        browser: {
            _responsiveState: true,
            breakpoints: defaultBreakpoints
        }
    })
    // create a redux store with the reducer at the root
    const store = createStore(
        reducers,
        preloadedState
    )

    // make sure we could retrieve the default breakpoints from the store
    expect(getBreakpoints(store)).toEqual(defaultBreakpoints)
})
```

Reply:
Ah, ok. One issue in your case is that `redux-responsive`'s reducer creates and expects plain-object state for the branch it manages, so converting your entire preloadedState to Immutable may cause you some issues, because you're ending up with a different data structure in your state than this lib expects.

I don't know what the best solution is here in your case, but to me it seems like `redux-responsive`'s branch of state should be left as-is and not converted to Immutable by users, since its reducer is built around plain objects. Users of this lib shouldn't have to change or worry about the data structure of the `redux-responsive` branch of state, it should just work as expected in a self-contained way, regardless of how the rest of their state is structured. Otherwise this lib is going to have to start managing some difficult edge cases (e.g. the extra type checking required to see if a user has transformed the state). What do you think @AlecAivazis?

(Side note: complications with 3rd-party libs like this are one of the reasons I stopped using Immutable.js, and started just using plain JS with Ramda immutability :smile:)

Reply:
Yea I agree @shanecav. I'm a bit wary of having to support a bunch of different edge cases. Although Immutable is definitely the most popular, this feels a bit like a slippery slope as new libraries show up. Like you said, users of `redux-responsive` shouldn't have to worry too much since the implementation is isolated from the API.

@bguggie I think the best option for you is to make sure that the responsive state is a vanilla object when hydrating the store from the backend.

Reply:
@shanecav @AlecAivazis ... I don't disagree that having `redux-responsive` handle `immutable` at all, is probably not the best pattern.  I initially saw that it had 'some' support and was surprised and was trying to contribute to help it work better.

I will refactor my usage to ensure `redux-responsive` gets passed a vanilla object. 

We can probably close this PR.   Thanks for humoring me.

Reply:
I really appreciate the work you put on this. I don't use immutable so I'm very happy that we explored this so thoroughly. I'm sure this question will come up again so I'm going to add something to the README describing this in the SSR section. Thanks so much!

And thanks to @shanecav for helping too!